### ISSUE TITLE
Consider switching to a real monorepo

### ISSUE THREAD

**Issue by jelhan** [ember-bootstrap/ember-bootstrap#1304]
This repository already contains two projects:

1. The `ember-bootstrap` package itself in the root.
2. The documentation app in `docs/` folder.

This is very confusing. Especially as both of them have their own dependencies. I just struggled myself with starting the docs app until I realized that I need to run `yarn install` within `docs/` folder. This is also preventing us from running dummy app and tests at the same time as documentation app.

I think it will be easier to maintain and less confusing to other developers if we switch to a classic monorepo file system layout:

```
- package.json (for yarn monorepo)
- yarn.lock
- packages/
  - ember-bootstrap/
  - docs/
```

Additionally this will allow us to pull official addons into this repository itself. This has some benefits like easier testing if an addon is affected by a change in Ember Bootstrap. But it also has some trade-offs as issues and pull requests are not separated anymore and tests may be run for packages, which aren't changed. I think we should discuss it separately. But wanted to highlight that it would *enable* us to do so.

**Comment by cah-briangantzler** [comment]
One note for ember addons, if you adopt the mono repo approach, you will no longer be able to point an application to a branch or commit for testing. ie you can not do this in a package.json

    "ember-cli-mirage": "git+https://github.com/miragejs/ember-cli-mirage#de8a248",

Many times this is required to allow someone to test a pending change without actually releasing a version

**Comment by drauschenbach** [comment]
I think you're misusing the term [monorepo](https://en.wikipedia.org/wiki/Monorepo), since you are advocating against multiple projects in the same repo. I almost always favor decomposition as you do as well.

**Comment by jelhan** [comment]
> One note for ember addons, if you adopt the mono repo approach, you will no longer be able to point an application to a branch or commit for testing. ie you can not do this in a package.json
> 
> ```
> "ember-cli-mirage": "git+https://github.com/miragejs/ember-cli-mirage#de8a248",
> ```
> 
> Many times this is required to allow someone to test a pending change without actually releasing a version

Very valid point. Maybe we should not change filesystem layout but use yarn workspaces to share the dependencies and only have one install step? :thinking: But to be honest I'm not sure yarn workspaces support to have a package at the workspace root. Would need to test that out.

**Comment by cah-briangantzler** [comment]
Since this seems to be the issue with the docs folder being a separate application, perhaps moving the docs into the tests/dummy application via ember-cli-addon-docs might be the solution

**Comment by simonihmig** [comment]
> Since this seems to be the issue with the docs folder being a separate application, perhaps moving the docs into the tests/dummy application via ember-cli-addon-docs might be the solution

It was like this for many years, but this had a ton of problems, which is why we moved it to a separate app! See https://github.com/kaliber5/ember-bootstrap/issues/929

**Comment by SanderKnauff** [comment]
This implemented in #2073.

Is there anything that should be done before closing this?

**Comment by jelhan** [comment]
I think we can consider it done. Thanks a lot (again) for picking it up!
